### PR TITLE
Small UX improvements

### PR DESCRIPTION
**Pull request by abdolence**
- Human predictable sorting for Project IDs and roles. 
- Adjusting UI stepper width when custom roles are used

image

### CODE REVIEW

File: sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java [hunk 1]
```diff
@@ -353,7 +353,7 @@ public Set<UserId> listEligibleUsersForProjectRole(
 
       // Remove the caller.
       .filter(user -> !user.equals(callerUserId))
-      .collect(Collectors.toSet());
+      .collect(Collectors.toCollection(TreeSet::new));
```
Comment (jpassing) [inline sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java:356]: Same here... `User` doesn't implement `Comparable` yet.

File: sources/src/main/java/com/google/solutions/jitaccess/core/adapters/ResourceManagerAdapter.java [hunk 2]
```diff
@@ -287,7 +287,7 @@ public List<ProjectId> searchProjectIds(String query) throws NotAuthenticatedExc
 
       return allProjects.stream()
         .map(p -> new ProjectId(p.getProjectId()))
-        .collect(Collectors.toList());
+        .collect(Collectors.toCollection(TreeSet::new));
```
[the remark this answers is not in the document]
Comment (abdolence) [reply sources/src/main/java/com/google/solutions/jitaccess/core/adapters/ResourceManagerAdapter.java:290]: Damn, I forgot to add some changes to the PR, sorry. Let me fix it.

File: sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java [hunk 3]
```diff
@@ -297,7 +297,7 @@ public Result<ProjectRole> listEligibleProjectRoles(
       Stream.ofNullable(analysisResult.getNonCriticalErrors())
         .flatMap(Collection::stream)
         .map(e -> e.getCause())
-        .collect(Collectors.toSet()));
+        .collect(Collectors.toCollection(TreeSet::new)));
```
[the remark this answers is not in the document]
Comment (abdolence) [reply sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java:300]: Yes, this is probably need to be returned back.

**Comment by abdolence** [comment]
@jpassing Sorry for confusion with TreeSet without Comparable.

Updated the PR with tests as well to cover this.

File: sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java [hunk 4]
```diff
@@ -291,9 +291,7 @@ public Result<ProjectRole> listEligibleProjectRoles(
     consolidatedRoles.addAll(activatedRoles);
```
Comment (jpassing) [inline sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java:291]: `consolidatedRoles` is only partially sorted now:

1. It contains JIT-eligible roles first, then MPA-eligible roles, and finally activated roles.
2. Within each of these 3 "buckets", the order is undefined.

(1) is good, but I guess you want each "bucket" to be sorted too. For that, you'd need to change the code above where `mpaEligibleRoles`, `jitEligibleRoles` and `activatedRoles` are populated.

Comment (abdolence) [reply sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java:291]: Thanks, addressed the comment!

File: sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java [hunk 5]
```diff
@@ -291,9 +291,7 @@ public Result<ProjectRole> listEligibleProjectRoles(
...
 
     return new Result<>(
-      consolidatedRoles.stream()
-        .sorted(Comparator.comparing(r -> r.roleBinding.fullResourceName))
-        .collect(Collectors.toList()),
+      consolidatedRoles.stream().sorted().collect(Collectors.toList()),
```
Comment (jpassing) [inline sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java:294]: The `sorted()` call destroys the order again, you probably have to remove this.

This is how it looks now -- it's sorted across all 3 buckets as opposed to within each bucket:
image

Comment (abdolence) [reply sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java]: Fixed. Yep, now that additional sorting can be removed as I understand.

**Comment by jpassing** [comment]
Looks good now:

image

Thanks for creating this PR!

**Pull request merged**